modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/ExpressionParser.java

```diff
@@ -307,24 +307,47 @@ public static ExpressionExecutor parseExpression(Expression expression, MetaComp
         } else if (expression instanceof Variable) {
             return parseVariable((Variable) expression, metaEvent, currentState, executorList, defaultStreamEventIndex);
 
+        } else if (expression instanceof IncrementalWithinTime) {
```

Comment:
If it's a function, we have to have a corresponding function executor. To use a function executor we need to add annotations and ultimately, these functions would be made available in the documentation as well. Since this is actually used for internal implementation only, users would be confused if this functionality is shown in our docs. Therefore, thought not to make this a function